src/libraries/Microsoft.Extensions.DependencyInjection/src/Microsoft.Extensions.DependencyInjection.csproj

```diff
@@ -5,6 +5,10 @@
     <!-- Debug IL generation -->
     <ILEmitBackendSaveAssemblies>False</ILEmitBackendSaveAssemblies>
     <Nullable>Annotations</Nullable>
+    <NoWarn>$(NoWarn);CP0001</NoWarn>
+<!--
+  'Microsoft.Extensions.DependencyInjection.ServiceCollection' exists on the previous stable version but not on the recent version.
```

Comment:
I checked and this type is no longer present in .NET 6 cc @ericstj

Reply:
It’s been forwarded down, we need to follow the typeforward.  That’s not possible in the current design that doesn’t have dependencies.  Perhaps we need a different diagnostic ID?  We could see that the type was replaced with a forward we just can’t compare the forwarded type without dependencies present.

Reply:
That is a good idea. We could do that in MemberMustExists tule when checking if the type exists and if it doesn't look for a typeforward for it in the other side and warn with a different diagnostic.

Reply:
yeah i can take a shot at implementing this. We will probably need this before enabling this for system.* because we might have more cases there. For here i will create an issue and edit the comment message

Reply:
> We could see that the type was replaced with a forward we just can’t compare the forwarded type without dependencies present.

do u mean to compare the fully qualified for the type on the left and typeforward name on the right ?

Reply:
I think it would be here? https://github.com/dotnet/sdk/blob/badd5e2cb36e8744b0dda485cecb26e7adbdb2c7/src/Compatibility/Microsoft.DotNet.ApiCompatibility/Rules/MembersMustExist.cs#L38

Another thought would be that we could add unresolved type object when we encounter an unresolved type forward:
https://github.com/dotnet/sdk/blob/badd5e2cb36e8744b0dda485cecb26e7adbdb2c7/src/Compatibility/Microsoft.DotNet.ApiCompatibility/Abstractions/Mappers/AssemblyMapper.cs#L112

That could let us consider the type present just not diff the members.  We could emit the diagnostic at the point we can't resolve the forward, and then halt the visitor when we encounter the unresolved type.